notebooks/204-document-entity-extraction/204-document-entity-extraction.ipynb

```diff
@@ -0,0 +1,615 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Imports"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "import cv2\n",
+    "import time\n",
+    "import json\n",
+    "import operator\n",
+    "from urllib import parse\n",
+    "\n",
+    "import pdfplumber\n",
+    "import pytesseract\n",
+    "\n",
+    "import numpy as np\n",
+    "import tokens_bert as tokens\n",
+    "from openvino.runtime import Core"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Preparations: Test tesseract OCR with sample image"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# On Windows, this script adds the directory that contains tesseract.exe to the PATH to enable OCR engine to find the\n",
+    "# required Python tools. This code assumes that tesseract python package is installed in the default\n",
+    "# directory. If you have a different installation path, please set the correct path in pytesseract.pytesseract.tesseract_cmd\n",
+    "\n",
+    "# https://stackoverflow.com/questions/50655738/how-do-i-resolve-a-tesseractnotfounderror\n",
+    "\n",
+    "import sys\n",
+    "\n",
+    "if sys.platform == \"win32\":\n",
+    "    import distutils.command.build_ext\n",
+    "    import os\n",
+    "    from pathlib import Path\n",
+    "\n",
+    "    OCR_INSTALL_DIR = r\"C:/Program Files/Tesseract-OCR\"\n",
+    "    ocr_path = sorted(list(Path(OCR_INSTALL_DIR).glob(\"**/tesseract.exe\")))\n",
+    "    if len(ocr_path) == 0:\n",
+    "        raise ValueError(\n",
+    "            \"Cannot find Tesseract OCR tool executable. This notebook requires setting \"\n",
+    "            \"pytesseract.pytesseract.tesseract_cmd to the directory that contains tesseract.exe \"\n",
+    "        )\n",
+    "    else:\n",
+    "        pytesseract.pytesseract.tesseract_cmd=str(ocr_path[-1])\n",
+    "        print(f\"Successfully set pytesseract.pytesseract.tesseract_cmd to path {ocr_path[-1]}\\n\")\n",
+    "        \n",
+    "    # Test converting sample JPG image into string using pytesseract OCR\n",
+    "    print('Extracted OCR text:')\n",
+    "    img = cv2.imread('data/sample.jpg')\n",
+    "    print(pytesseract.image_to_string(img))"
```

Comment:
These lines should be under else to deal with the situation you don't have tesseract installed on Windows